warehouse/scripts/publish.py

```diff
@@ -477,14 +477,19 @@ def _publish_exposure(
 
                     if publish:
                         typer.secho(publish_msg, fg=typer.colors.GREEN)
-                        with open(fpath, "rb") as fp:
-                            upload_to_ckan(
-                                url=destination.url,
-                                fname=fname,
-                                fsize=fsize,
-                                file=fp,
-                                resource_id=resource.id,
-                            )
+                        try:
+                            with open(fpath, "rb") as fp:
+                                upload_to_ckan(
+                                    url=destination.url,
+                                    fname=fname,
+                                    fsize=fsize,
+                                    file=fp,
+                                    resource_id=resource.id,
+                                )
+                        except requests.exceptions.RequestException as e:
```

Comment:
I believe that this should be `requests.exceptions.HTTPError`, which is the specific class of error produced [when an upload times out](https://o1d2fa0877cf3fb10p-tp.appspot.com/log?dag_id=publish_open_data&task_id=publish_california_open_data&execution_date=2024-01-01T00%3A00%3A00%2B00%3A00&map_index=-1).

Reply:
yes this makes sense! I almost did that but defaulted to a more general exception for some unknown reason